File: Java/Eta/Applications/Shared/src/main/java/com/thomsonreuters/upa/shared/provider/ItemRequestInfo.java
package com.thomsonreuters.upa.shared.provider;

import com.thomsonreuters.upa.codec.Buffer;
import com.thomsonreuters.upa.codec.CodecFactory;
import com.thomsonreuters.upa.codec.MsgKey;
import com.thomsonreuters.upa.transport.Channel;

/**
 * Item request information.
 */
public class ItemRequestInfo
{
    public int streamId;
    public Buffer itemName;
    public ItemInfo itemInfo;
    public boolean isStreamingRequest;
    public boolean isPrivateStreamRequest;
    public boolean includeKeyInUpdates;
    public MsgKey msgKey;
    public Channel channel;
    public boolean isInUse;
    public int domainType;

    /**
     * Instantiates a new item request info.
     */
    public ItemRequestInfo()
    {
        streamId = 0;
        itemName = CodecFactory.createBuffer();
        itemInfo = null;
        isStreamingRequest = false;
        isPrivateStreamRequest = false;
        includeKeyInUpdates = false;
        msgKey = CodecFactory.createMsgKey();
        channel = null;
        isInUse = false;
        domainType = 0;
    }

    /**
     * Clear.
     */
    public void clear()
    {
        streamId = 0;
        itemName.clear();
        itemInfo = null;
        isStreamingRequest = false;
        isPrivateStreamRequest = false;
        includeKeyInUpdates = false;
        msgKey.clear();
        channel = null;
        isInUse = false;
        domainType = 0;
    }
    
    /**
     * Stream id.
     *
     * @return the int
     */
    public int streamId()
    {
    	return streamId;
    }
    
    /**
     * Stream id.
     *
     * @param streamId the stream id
     */
    public void streamId(int streamId)
    {
    	this.streamId = streamId;
    }
    
    /**
     * Item name.
     *
     * @return the buffer
     */
    public Buffer itemName()
    {
    	return itemName;
    }
    
    /**
     * Item info.
     *
     * @return the item info
     */
    public ItemInfo itemInfo()
    {
    	return itemInfo;
    }
    
    /**
     * Item info.
     *
     * @param itemInfo the item info
     */
    public void itemInfo(ItemInfo itemInfo)
    {
    	this.itemInfo = itemInfo;
    }
    
    /**
     * Checks if is streaming request.
     *
     * @return true, if is streaming request
     */
    public boolean isStreamingRequest()
    {
    	return isStreamingRequest;
    }
    
    /**
     * Checks if is streaming request.
     *
     * @param isStreamingRequest the is streaming request
     */
    public void isStreamingRequest(boolean isStreamingRequest)
    {
    	this.isStreamingRequest = isStreamingRequest;
    }
    
    /**
     * Checks if is private stream request.
     *
     * @return true, if is private stream request
     */
    public boolean isPrivateStreamRequest()
    {
    	return isPrivateStreamRequest;
    }
    
    /**
     * Checks if is private stream request.
     *
     * @param isPrivateStreamRequest the is private stream request
     */
    public void isPrivateStreamRequest(boolean isPrivateStreamRequest)
    {
    	this.isPrivateStreamRequest = isPrivateStreamRequest;
    }
    
    /**
     * Include key in updates.
     *
     * @return true, if successful
     */
    public boolean includeKeyInUpdates()
    {
    	return includeKeyInUpdates;
    }
    
    /**
     * Include key in updates.
     *
     * @param includeKeyInUpdates the include key in updates
     */
    public void includeKeyInUpdates(boolean includeKeyInUpdates)
    {
    	this.includeKeyInUpdates = includeKeyInUpdates;
    }
    
    /**
     * Msg key.
     *
     * @return the msg key
     */
    public MsgKey msgKey()
    {
    	return msgKey;
    }
    
    /**
     * Channel.
     *
     * @return the channel
     */
    public Channel channel()
    {
    	return channel;
    }

    /**
     * Channel.
     *
     * @param channel the channel
     */
    public void channel(Channel channel)
    {
    	this.channel = channel;
    }

    /**
     * Checks if is in use.
     *
     * @return true, if is in use
     */
    public boolean isInUse()
    {
    	return isInUse;
    }
    
    /**
     * Checks if is in use.
     *
     * @param isInUse the is in use
     */
    public void isInUse(boolean isInUse)
    {
    	this.isInUse = isInUse;
    }
    
    /**
     * Domain type.
     *
     * @return the int
     */
    public int domainType()
    {
    	return domainType;
    }

    /**
     * Domain type.
     *
     * @param domainType the domain type
     */
    public void domainType(int domainType)
    {
    	this.domainType = domainType;
    }
}

File: CSharp/Ema/Src/Tests/OmmConsumerTests/ItemRequestInfo.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2023 LSEG. All rights reserved.     
 *|-----------------------------------------------------------------------------
 */

using System;
using LSEG.Eta.Codec;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Buffer = LSEG.Eta.Codec.Buffer;
using LSEG.Eta.ValueAdd.Reactor;

namespace LSEG.Ema.Access.Tests.OmmConsumerTests
{
    internal class ItemRequestInfo
    {
        public int StreamId { get; set; }

        /// <summary>
        /// Gets itema name
        /// </summary>
        public Buffer ItemName { get; private set; } = new Buffer();

        /// <summary>
        /// Gets or sets whether if is streaming request
        /// </summary>
        public bool IsStreamingRequest { get; set; }

        /// <summary>
        /// Gets or sets whether if is private stream request
        /// </summary>
        public bool IsPrivateStreamRequest { get; set; }

        /// <summary>
        /// Gets or sets whether include key in updates
        /// </summary>
        public bool IncludeKeyInUpdates { get; set; }

        /// <summary>
        /// Gets message key
        /// </summary>
        public IMsgKey MsgKey { get; set; } = new MsgKey();


        /// <summary>
        /// Gets or sets <c>ReactorChannel</c>
        /// </summary>
        public ReactorChannel? ReactorChannel { get; set; }

        /// <summary>
        /// Gets or sets if is in use.
        /// </summary>
        public bool IsInUse { get; set; }

        /// <summary>
        /// Gets or sets domain type
        /// </summary>
        public int DomainType { get; set; }

        /// <summary>
        /// Instantiates a new item request info.
        /// </summary>
        public ItemRequestInfo()
        {
            StreamId = 0;
            IsStreamingRequest = false;
            IsPrivateStreamRequest = false;
            IncludeKeyInUpdates = false;
            ReactorChannel = null;
            IsInUse = false;
            DomainType = 0;
        }

        /// <summary>
        /// Clears to default values
        /// </summary>
        public void Clear()
        {
            StreamId = 0;
            ItemName.Clear();
            IsStreamingRequest = false;
            IsPrivateStreamRequest = false;
            IncludeKeyInUpdates = false;
            MsgKey.Clear();
            ReactorChannel = null;
            IsInUse = false;
            DomainType = 0;
        }
    }
}
